agir/front/api_views.py

```diff
@@ -100,15 +99,29 @@ def get_events(self, search_term, filters, result_limit=20):
             events = events.filter(location_country=country)
         if eventType:
             events = events.filter(subtype__type=eventType)
+
         if eventCategory:
             if eventCategory == self.EVENT_FILTER_PAST:
                 events = events.filter(end_time__lte=timezone.now())
             else:
                 events = events.filter(end_time__gte=timezone.now())
+        else:
+            # Default: get only upcoming events
+            eventsQueryFiltered = events
+            events = events.filter(end_time__gte=timezone.now())
 
         # Query
         events = events.search(search_term).distinct()
 
+        # Default: fill with past events if results < result_limit
+        if not eventCategory:
+            countFutureEvents = events.count()
+            if countFutureEvents < result_limit:
+                events_past = eventsQueryFiltered.filter(end_time__lte=timezone.now())[
+                    : (result_limit - countFutureEvents)
+                ]
+                events = events | events_past
+
```

Comment:
Est-ce que ça pourrait suffire comme ça, tu crois ? (pour éviter trop de requêtes)
```suggestion
        # Query
        events = events.search(search_term).distinct()

        if not eventCategory and events.upcoming().count() >= result_limit:
          events = events.upcoming()
```

Reply:
Non, en vrai j'ai l'impression que le nombre de requêtes serait le même dans les deux cas : une requête `count` et une requête pour récupérer les résultats de la recherche. Personnellement je trouve bizarre d'ajouter une condition WHERE (ligne 111) 
```suggestion
        else:
            # Default: get only upcoming events
            eventsQueryFiltered = events
            events = events.filter(end_time__gte=timezone.now())

        # Query
        events = events.search(search_term).distinct()

        # Default: fill with past events if results < result_limit
        if not eventCategory and events.count() < result_limit:
                events_past = eventsQueryFiltered.filter(end_time__lte=timezone.now())[
                    : (result_limit - countFutureEvents)
                ]
                events = events | events_past

```

Reply:
Ah si au temps pour moi, je ne comptais plus mon `count` du coup ! :smile: